### ISSUE TITLE
Package up maze generation impls (and Ariadne's zipper?) into a project (Chapter 3)

### ISSUE THREAD

**Issue by cemerick** [clojurebook/ClojureProgramming#7]
@cgrand, maybe you'd like to take care of this?

**Comment by rogerallen** [comment]
Not sure if this is the appropriate place or not, but I was creating a project based on the Ariadne zipper.

I ran into one issue that you might want to consider.  Sometimes, two points cannot be connected.  In this case, the zipper hangs.

Is there a way for the zipper to detect this hang situation and return an empty path?

**Comment by cgrand** [comment]
The Ariadne zipper only works on connected acyclic graphs — and the generated mazes are such graphs. 
I'm pretty confident it can be adapted to work on larger (if not all) classes of graphs. 
What can you tell me about your graph?

**Comment by rogerallen** [comment]
I'm not creating my own graph, I'm just trying to use the code from the book.  I created a gist here:

https://gist.github.com/2715291

I edited the code as little as possible from the text here:
https://github.com/clojurebook/ClojureProgramming/blob/master/ch03-collections-repl-interactions.clj
to get it to run and reproduce the issue.  I tried to comment right where i modified the code.  I can't figure it out...but I'm new to Clojure, as I just got the book this week.

If you load a repl via "java -jar clojure-1.3.0.jar", paste in the gist text, then invoke (testit) multiple times, it hangs relatively often for me (1 of 4 times at least).  When it hangs, the maze is drawn (via my added first call with empty path), the println for the location of theseus & minotaur is output, but no path is drawn.  I assume that the calculation of the path is hanging.

**Comment by cgrand** [comment]
Good catch: if the minotaur is in a dead-end, its location never appears in the value returned by z/path. It does only affect the version with the filter not the first one which just prints the path.

**Comment by cgrand** [comment]
@rogerallen fixed here https://github.com/clojurebook/ClojureProgramming/commit/9ffe8d4374791e577b6a4e2417efc544fecc8126
